src/app/profile/profile.page.ts

```diff
@@ -1,23 +1,47 @@
 import { Component, OnInit } from '@angular/core';
+import { Router, NavigationExtras, ActivatedRoute } from '@angular/router';
 
 @Component({
   selector: 'app-profile',
   templateUrl: './profile.page.html',
   styleUrls: ['./profile.page.scss'],
 })
 export class ProfilePage implements OnInit {
-  public user: any;
-  constructor() {
-    this.user = {
+
+  user;
+
+  constructor(private route: ActivatedRoute, private router: Router) {
+    console.log("Profile page started (constructor)");
+    var defaultUser = {
       name : "Burnest Griffin IV",
       company : "IdeaLogic",
       role : "Developer"
     };
 
-    console.log("Profile page started");
+    this.route.data.subscribe((data) => {
```

Comment:
This promise only jogs things along after a submission. It doesn't matter if we watch `data`, `params`, or `queryParams`.

Reply:
That's good to know